### PR TITLE
Ncs 632 costs link

### PR DESCRIPTION
**Pull request by mattch1**
Call oracle query api to get a flag to say whether the confirmation statement is paid for a given period for that company. If true we do not add a costs link to the transaction, if it is false we add this second link.

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/confirmationstatementapi/client/OracleQueryClient.java [hunk 1]
```diff
@@ -113,11 +114,26 @@ public ActiveDirectorDetails getActiveDirectorDetails(String companyNumber) thro
...
+       var paymentsUrl = String.format(
+               "%s/company/%s/confirmation-statement/paid/?payment_period_due_date=%s", oracleQueryApiUrl, companyNumber, dueDate);
+
+        LOGGER.info(CALLING_ORACLE_QUERY_API_URL_GET, paymentsUrl);
+        ResponseEntity<Boolean> response = restTemplate.getForEntity(paymentsUrl, Boolean.class);
+        if (response.getStatusCode() != HttpStatus.OK) {
+            throw new ServiceException(String.format(ORACLE_QUERY_API_STATUS_MESSAGE, response.getStatusCode(), companyNumber));
+        }
+        if (response.getBody() == null) {
+            throw new ServiceException("Oracle query api returned null, boolean values expected");
```
Comment (markpit) [inline src/main/java/uk/gov/companieshouse/confirmationstatementapi/client/OracleQueryClient.java:135]: I'd also add the due date into the 2 service exception messages, it'll help us to identify the payment periods when these error are thrown in Live.

File: src/test/java/uk/gov/companieshouse/confirmationstatementapi/client/OracleQueryClientTest.java [hunk 2]
```diff
@@ -181,4 +184,32 @@ void testGetShareholderNotOkStatusResponse() {
         assertTrue(serviceException.getMessage().contains(companyNumber));
         assertTrue(serviceException.getMessage().contains(HttpStatus.SERVICE_UNAVAILABLE.toString()));
     }
+
+    @Test
+    void testIsConfirmationStatementPaid() throws ServiceException {
+        ResponseEntity<Boolean> response = ResponseEntity.status(HttpStatus.OK).body(Boolean.TRUE);
```
Comment (markpit) [inline src/test/java/uk/gov/companieshouse/confirmationstatementapi/client/OracleQueryClientTest.java:190]: I don't think we are allowed to return just a value in the response that isn't attached to a field that indicates what it represents. As far as I know all CHS api return field: value  in the response. I'll tag @ICoxCH and @sroberts2 for advice but looks like it's something I missed in the review of the oracle-query-api paid endpoint.

Comment (mattch1) [reply src/test/java/uk/gov/companieshouse/confirmationstatementapi/client/OracleQueryClientTest.java:190]: I have tested this and it works as expected. Not allowed by who?

Comment (markpit) [reply src/test/java/uk/gov/companieshouse/confirmationstatementapi/client/OracleQueryClientTest.java:190]: I'll raise it separately and get others views as it doesn't really affect this pr , i wasn't sure if the oracle-query-api response was inline with the other CHS apis ie. whether the responses need to be wrapped in a json object or if we can return the primitive value

**Review by markpit — APPROVED** [review]

File: src/main/java/uk/gov/companieshouse/confirmationstatementapi/client/OracleQueryClient.java [hunk 3]
```diff
@@ -113,11 +115,27 @@ public ActiveDirectorDetails getActiveDirectorDetails(String companyNumber) thro
...
         }
         if (response.getBody() == null) {
             return new ArrayList<>();
         }
         return Arrays.asList(response.getBody());
     }
+
+    public boolean isConfirmationStatementPaid(String companyNumber, String dueDate) throws ServiceException {
+       var paymentsUrl = String.format(
+               "%s/company/%s/confirmation-statement/paid/?payment_period_due_date=%s", oracleQueryApiUrl, companyNumber, dueDate);
```
Comment (markpit) [inline src/main/java/uk/gov/companieshouse/confirmationstatementapi/client/OracleQueryClient.java:128]: "%s/company/%s/confirmation-statement/paid/?payment_period_due_date=%s"

the '/' after paid and before the ? is not needed, can be removed

**Review by ICoxCH — APPROVED** [review]

**Pull request merged**